src/MTProtoTools/FilesAbstraction.php

```diff
@@ -132,6 +143,9 @@ public function wrapMedia(array $media, bool $protected = false): ?Media
         if ($has_document_photo) {
             return new DocumentPhoto($this, $media, $has_document_photo, $protected);
         }
+        if ($media['_'] === 'decryptedMessageMediaDocument') {
+            return new Media\Decrypted\Document($this, $media, $protected);
```

Comment:
I think that creating separate classes for secret media is not the way to go, instead we could just add some fields to the existing media classes that will only be populated in secret chats (i.e. fields that are not present for normal media)